homeassistant/components/hue/hue_sensor.py

```diff
@@ -0,0 +1,237 @@
+"""Support for the Philips Hue sensors as a platform."""
+import asyncio
+from datetime import timedelta
+import logging
+from time import monotonic
+
+import async_timeout
+
+from homeassistant.components import hue
+from homeassistant.helpers.event import async_track_point_in_utc_time
+from homeassistant.util.dt import utcnow
+
+
+DEPENDENCIES = ['hue']
+SCAN_INTERVAL = timedelta(seconds=5)
+
+PRESENCE_NAME_FORMAT = "{} presence"
+LIGHT_LEVEL_NAME_FORMAT = "{} light level"
+IS_DARK_NAME_FORMAT = "{} is not dark"
+TEMPERATURE_NAME_FORMAT = "{} temperature"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities,
+                            binary=False):
+    """Set up the Hue sensors from a config entry."""
+    bridge = hass.data[hue.DOMAIN][config_entry.data['host']]
+    cur_sensors = {}
```

Comment:
Why not store this in `hass.data` so that you can share it between sensors and binary sensors ?

Reply:
Good idea